### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by butlermuwo**
## validations, model specs, and n+1 problems.
In this pull request I did the following:
- I added validations for the User model.
- I added validations for the Post model.
- Added unit specs for all of my models, methods, and validations.
- Added flash messages in the create actions in all my controllers.
- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -1,5 +1,47 @@
 require 'rails_helper'
 
 RSpec.describe Comment, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'Validations For the Comment model' do
```
Comment (Deepakdanger) [inline spec/models/comment_spec.rb]: - [x] Kindly use some positive test cases, for the `comment` model test.

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
 
 RSpec.describe Like, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'Validations For the Like model' do
+    before(:each) do
+      @like = Like.new(author_id: 2, post_id: 7)
+    end
+
+    before { @like.save }
+
```
Comment (Deepakdanger) [inline spec/models/like_spec.rb:23]: - [x] Kindly use some positive test cases, for the `like` model test.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @butlermuwo, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (butlermuwo) [reply spec/models/comment_spec.rb]: Hey @Deepakdanger please show me an example of how to implement positive test cases?

File: spec/models/like_spec.rb (hunk 2 above)
Comment (butlermuwo) [reply spec/models/like_spec.rb:23]: Hey @Deepakdanger please show me an example of how to implement positive test cases?

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (butlermuwo) [reply spec/models/comment_spec.rb]: I have Implemented it even though you are not willing to help and is not part of the requirements.

File: spec/models/like_spec.rb (hunk 2 above)
Comment (butlermuwo) [reply spec/models/like_spec.rb:23]: Thanks @Deepakdanger

**Review by DeliceLydia — APPROVED** [review]
## Hi @butlermuwo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by butlermuwo** [comment]
Thanks @DeliceLydia 😊🙌🏾

**Pull request merged**